src/vstorage-query.ts

```diff
@@ -1,7 +1,67 @@
-import { abciQuery } from './abci-query';
+import { timeoutDurationMS } from './constants';
 
-export const vstorageQuery = async (
-  node: string,
-  path: string,
-  blockHeight?: number
-) => abciQuery(node, `/custom/vstorage/data/${path}`, blockHeight);
+const getNodeIterator = (nodes: string[]) => {
```

Comment:
this reads like it's an Iterator but it's just different enough to be confusing. Please return an actual [JS Iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols)

```js
interface IteratorReturnResult<TReturn> {
    done: true;
    value: TReturn;
}
interface Iterator<T, TReturn = any, TNext = undefined> {
    // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
    next(...args: [] | [TNext]): IteratorResult<T, TReturn>;
    return?(value?: TReturn): IteratorResult<T, TReturn>;
    throw?(e?: any): IteratorResult<T, TReturn>;
}
```

```suggestion
const getNodeIterator = (nodes: string[]): Iterator<string> => {
```

Reply:
FWIW, there's also a convention in SwingSet (albeit a weak one) to prefix the name of such functions with "enumerate". And in this case also I think "node" is misleading because there are both RPC nodes and storage nodes(neither of which are relevant to the implementation), and more substantially that there might be benefit in mixing up the order in each block of the nodes. Suggestion:
```typescript
/**
 * Enumerates an infinite sequence composed of elements from
 * successive Fisher-Yates shuffles of an array.
 *
 * @param elements
 * @returns An iterator
 */
const enumerateShuffles = <T,>(elements: T[]): Iterator<T> => {
  const indices = elements.map((_element, i) => i);
  // https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
  const shuffleIndices = () => {
    for (let i = indices.length - 1; i >= 1; i -= 1) {
      const j = Math.floor(Math.random() * (i + 1));
      const tmp = indices[j];
      indices[j] = indices[i];
      indices[i] = tmp;
    }
  };

  let i = elements.length;
  const next: () => IteratorResult<T> = () => {
    i += 1;
    if (i >= elements.length) {
      shuffleIndices();
      i = 0;
    }
    return { done: false, value: elements[indices[i]] };
  };
  return { next };
};
```

Reply:
Incorporated @gibson042's suggestion